GNSSLogger/app/src/test/java/com/google/android/apps/location/gps/gnsslogger/PendingIntentCompatTest.java

```diff
@@ -0,0 +1,90 @@
+package com.google.android.apps.location.gps.gnsslogger;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.robolectric.Shadows.shadowOf;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+import androidx.test.core.app.ApplicationProvider;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+import org.robolectric.shadows.ShadowPendingIntent;
+
+/** Unit test for {@link PendingIntentCompat}. */
+@RunWith(RobolectricTestRunner.class)
+public class PendingIntentCompatTest {
+  private final Context context = ApplicationProvider.getApplicationContext();
+
+  /** Immutable is introduced in M, but the project's minimum support version is 24 */
+  @Config(sdk = Build.VERSION_CODES.N)
+  @Test
+  public void addMutabilityFlags_immutableOnMPlus() {
+    int requestCode = 7465;
+    Intent intent = new Intent();
+    Bundle options = new Bundle();
+    ShadowPendingIntent shadow =
+        shadowOf(
+            PendingIntentCompat.getBroadcast(
+                context,
+                requestCode,
+                intent,
+                /* isMutable= */ false,
+                PendingIntent.FLAG_UPDATE_CURRENT));
+    assertThat(shadow.getFlags())
+        .isEqualTo(PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
+  }
+
+  @Config(sdk = Build.VERSION_CODES.R)
+  @Test
+  public void addMutabilityFlags_mutableOnPreS() {

```

Comment:
> Given that the only branch for API level in the PendingIntentCompat is for `Build.VERSION_CODES.S`, please test for the following combinations (one method for each):
> 
> 1. `<` S, isMutable=true (`addMutabilityFlags_mutableOnPreS()`)
> 2. `<` S, isMutable=false (`addMutabilityFlags_immutableOnPreS()`)
> 3. `>=` S, isMutable=true  (`addMutabilityFlags_mutableOnSPlus()`)
> 4. `>=` S, isMutable=false (`addMutabilityFlags_immutableOnSPlus()`)
> 
> You can use `@SdkSupress` to help with this so the methods runs on multiple API levels.
> 
> For example:
> 
> `@SdkSupress(maxSdkVersion = Build.VERSION_CODES.R)` for 1 and 2
> 
> `@SdkSupress(minSdkVersion = Build.VERSION_CODES.S)` for 3 and 4
> 
> See https://developer.android.com/reference/androidx/test/filters/SdkSuppress.

Thanks. Ok, I will add these test. Since I've used Robolectric, `@Config` could be used.